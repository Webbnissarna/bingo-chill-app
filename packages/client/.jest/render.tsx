import React from "react";
import { render as rtlRender } from "@testing-library/react";

export function render(ui: React.ReactElement) {
  return rtlRender(<>{ui}</>);
}
