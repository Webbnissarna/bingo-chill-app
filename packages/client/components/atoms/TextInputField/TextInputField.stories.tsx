import { useState } from "react";
import TextInputField from ".";
import type { StoryDefault } from "@ladle/react";

export function Default() {
  const [value, setValue] = useState("Hello World");
  return (
    <div className="bg-polarNight-1 p-5">
      <TextInputField value={value} onChange={setValue} />
    </div>
  );
}

export default {
  title: "Atoms / Text Input Field",
} satisfies StoryDefault;
