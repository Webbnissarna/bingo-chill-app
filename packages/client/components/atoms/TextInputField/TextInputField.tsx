interface TextInputFieldProps {
  value: string;
  onChange: (newValue: string) => void;
}

export default function TextInputField({
  value,
  onChange,
}: TextInputFieldProps): JSX.Element {
  return (
    <input
      className="w-full font-text text-base text-snowStorm-2 outline-none border-none shadow-none bg-polarNight-0 rounded px-2 py-1 focus:bg-polarNight-2 transition"
      type="text"
      value={value}
      onChange={(e) => onChange(e.currentTarget.value)}
    />
  );
}
