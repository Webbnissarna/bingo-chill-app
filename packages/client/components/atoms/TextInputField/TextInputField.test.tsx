import { render } from "../../../.jest/render";
import TextInputField from ".";

describe("TextInputField", () => {
  it("renders", () => {
    const { asFragment } = render(
      <TextInputField value="Hello World" onChange={() => undefined} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
