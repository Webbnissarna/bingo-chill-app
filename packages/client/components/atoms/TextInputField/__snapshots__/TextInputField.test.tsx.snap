// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputField renders 1`] = `
<DocumentFragment>
  <input
    class="w-full font-text text-base text-snowStorm-2 outline-none border-none shadow-none bg-polarNight-0 rounded px-2 py-1 focus:bg-polarNight-2 transition"
    type="text"
    value="Hello World"
  />
</DocumentFragment>
`;
